### PR TITLE
Add Java build/test dependancies JUnit and Google Protobuf Java to cgmanifest

### PR DESCRIPTION
**Pull request by yuslepukhin**
**Description**: 
Add references to junit and google protobuf to cgmanifest

**Motivation and Context**
Legal requirements

### CODE REVIEW

**Review by pranavsharma — APPROVED** [review]

File: cgmanifests/submodules/cgmanifest.json [hunk 1]
```diff
@@ -100,6 +100,28 @@
         "comments": "manylinux dependency"
       }
     },
+    {
```
Comment (edgchen1) [inline cgmanifests/submodules/cgmanifest.json:103]: perhaps these should be added to cgmanifests/cgmanifest.json instead as they are not git submodules

File: cgmanifests/submodules/cgmanifest.json [hunk 2]
```diff
@@ -100,6 +100,28 @@
...
+      "Component": {
+        "Type": "other",
```
Comment (snnn) [inline cgmanifests/submodules/cgmanifest.json:105]: You may use a more concrete type, like
```json
"Component": {
"Type":  "maven",
"Maven":{ "GroupId": "junit", "ArtifactId": "junit", "Version": "4.13" }
}
```

File: cgmanifests/submodules/cgmanifest.json (hunk 1 above)
Comment (yuslepukhin) [reply cgmanifests/submodules/cgmanifest.json:103]: > perhaps these should be added to cgmanifests/cgmanifest.json instead as they are not git submodules

That file contains *only* submodules. The file I have updated contains `other` kinds of artifacts.

Comment (snnn) [reply cgmanifests/submodules/cgmanifest.json:103]: Indeed,  here "other" means the other things that we couldn't handle well.

Comment (edgchen1) [reply cgmanifests/submodules/cgmanifest.json:103]: cgmanifests/cgmanifest.json contains non-submodule entries as well. I suppose it's a bit mixed up now, but originally the intent was for cgmanifests/submodules/cgmanifest.json to contain git submodule entries, hence the "submodules" in the path.
https://github.com/microsoft/onnxruntime/blob/master/cgmanifests/README.md

**Review by edgchen1 — APPROVED** [review]

File: cgmanifests/cgmanifest.json [hunk 3]
```diff
@@ -1,5 +1,19 @@
 {
    "Registrations": [
+      {
+        "component": {
+          "Type": "maven",
+          "maven": { "GroupId": "org.junit.platform", "ArtifactId": "junit-platform-console-standalone", "Version": "1.6.2" }
```
Comment (snnn) [inline cgmanifests/cgmanifest.json:6]: Missing comma at the end of this line

File: cgmanifests/cgmanifest.json [hunk 4]
```diff
@@ -1,5 +1,19 @@
...
+          "DevelopmentDependency": true
+        }
+      },
+      {
+        "component": {
+          "Type": "maven",
+          "maven": { "GroupId": "com.google.protobuf", "ArtifactId": "protobuf-java", "Version": "3.9.2" }
```
Comment (snnn) [inline cgmanifests/cgmanifest.json:13]: Missing comma at the end of this line

**Review by snnn — APPROVED** [review]

File: cgmanifests/submodules/cgmanifest.json (hunk 1 above)
Comment (yuslepukhin) [reply cgmanifests/submodules/cgmanifest.json]: How do I auto-regenerate it? The script apparently handles only trivial GIT type entries.

Comment (edgchen1) [reply cgmanifests/submodules/cgmanifest.json]:
```
$ python cgmanifests/submodules/generate_submodule_cgmanifest.py > cgmanifests/submodules/cgmanifest.json
```
that should generate it.
it handles the project's git submodules and parses some dependencies from one of the dockerfiles

**Pull request merged**